examples/concurrency.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the MIT License, <LICENSE or http://opensource.org/licenses/MIT>.
+// This file may not be copied, modified, or distributed except according to those terms.
+
+#![plugin(serde_macros)]
+#![feature(custom_derive, plugin, default_type_parameter_fallback, test, try_from, inclusive_range_syntax)]
+
+extern crate chrono;
+extern crate env_logger;
+#[macro_use]
+extern crate log;
+extern crate mio;
+extern crate serde;
+#[macro_use]
+extern crate tarpc;
+extern crate test;
+
+use serde::{Serialize, Deserialize};
+use std::ops::Add;
+use std::time::{Duration, Instant, SystemTime, UNIX_EPOCH};
+use tarpc::{Client, RpcResult};
+
+service! {
+    rpc read(size: u32, sent_at: SerializableInstant) -> (Vec<u8>, SerializableInstant);
+}
+
+struct Server;
+
+impl SyncService for Server {
+    fn read(&self, size: u32, sent_at: SerializableInstant) -> RpcResult<(Vec<u8>, SerializableInstant)>  {
+        return Ok((gen_vec(size as usize), sent_at));
+
+        fn gen_vec(size: usize) -> Vec<u8> {
+            let mut vec: Vec<u8> = Vec::with_capacity(size);
+            for i in 0..size {
+                vec.push((i % 1 << 8) as u8);
+            }
+            vec
+        }
+    }
+}
+
+fn run_once(clients: &[FutureClient], concurrency: u32, print: bool) {
+    let _ = env_logger::init();
+
+    let start = Instant::now();
+    let futures: Vec<_> = 
+        clients.iter()
+               .cycle()
+               .take(concurrency as usize)
+               .map(|client| client.read(&CHUNK_SIZE, &SystemTime::now().serializable()))
```

Comment:
It's just part of the example. IMO it makes for a more interesting example to send the time as part of the rpc.